backend/src/composite/mod.rs

```diff
@@ -10,11 +16,16 @@ use crate::{Backend, BackendFactory, BackendOptions, Error, Proof};
 
 mod split;
 
+#[derive(Serialize, Deserialize)]
+struct CompositeVerificationKey {
+    verification_key: Vec<Vec<u8>>,
+}
+
 /// A composite proof that contains a proof for each machine separately.
 #[derive(Serialize, Deserialize)]
 struct CompositeProof {
     /// Map from machine name to proof
-    proofs: BTreeMap<String, Vec<u8>>,
+    proofs: Vec<Vec<u8>>,
```

Comment:
I changed it so that we don't have to store the strings, and get a more accurate estimate of the proof size. Same for the verification key.